README.md

```diff
@@ -137,12 +144,58 @@ Each *Node* annotation **MUST** be prefixed with:<br />
 Key | Values | Default | Description
 ---|---|---|---
 `private-ip` | `IPv4` | `none` | Specifies the Linode Private IP overriding default detection of the Node InternalIP.<br />When using a [VLAN] or [VPC], the Node InternalIP may not be a Linode Private IP as [required for NodeBalancers] and should be specified.
-
+`linode-firewall-id` | string | | An existing Cloud Firewall ID to be attached to the Node. See [Firewalls](#linode-firewalls).
+`linode-firewall-acl` | string | | The Firewall rules to be applied to the Node. Adding this annotation creates a new CCM-managed Linode CloudFirewall instance. See [Firewalls](#linode-firewalls).
 
 [required for NodeBalancers]: https://www.linode.com/docs/api/nodebalancers/#nodebalancer-create__request-body-schema
 [VLAN]: https://www.linode.com/products/vlan/
 [VPC]: https://www.linode.com/blog/linode/new-betas-coming-to-green-light/
 
+#### Linode Firewalls
+Firewall rules can be applied to the CCM-managed Nodes in two distinct ways.
+
+##### CCM-managed Firewall
+To use this feature, ensure that the Linode API token used with the CCM has the `add_firewalls` grant. 
+
+The CCM accepts firewall ACLs in JSON form. The ACL can either be an `allowList` or a `denyList`.
+Supplying both is not supported. Supplying neither is not supported. The `allowList` sets up a
+CloudFirewall that `ACCEPT`s traffic only from the specified IPs/CIDRs and `DROP`s everything else.
+The `denyList` sets up a CloudFirewall that `DROP`s traffic only from the specified IPs/CIDRs and
+`ACCEPT`s everything else. Ports are automatically inferred from the service configuration. 
+
+See [Firewall rules](https://www.linode.com/docs/api/networking/#firewall-create__request-body-schema)
+for more details on how to specify the IPs/CIDRs
+
+Example usage of an ACL to allow traffic from a specific set of addresses
+
+```yaml
+kind: Node
+apiVersion: v1
+metadata:
+  name: k8s-worker-1
+  annotations:
+    node.k8s.linode.com/linode-firewall-acl: |
+      {
+        "allowList": {
+          "ipv4": ["192.166.0.0/16", "172.23.41.0/24"],
+          "ipv6": ["2001:DB8::/128"]
+        },
+        "ports": ["22","6443"]
```

Comment:
is a user expected to supply ports? Can't we open up nodeports if a service is configured?

Reply:
Ports are needed for Node Firewalls specifically (e.g. access to k8s API by workers which does not involve NodePorts)

Reply:
is CAPL supplying this annotation, or is this going to be a user? trying to see if there's a way to make this more easier for users by making the ccm infer what the right thing to do is. 
1. hostPort services would need their ports opened
2. nodePort services would need ports opened (I need to double check if firewall covers the private ip, not needed if it doesn't)
3. open ssh port? 

IMO at-least #2 should be covered because we have all the info we need to do so

Reply:
My thought was CAPL would supply the annotation for the created nodes